tests/cross/src/main/scala/tests/BasePCSuite.scala

```diff
@@ -1,121 +1,203 @@
-      .addDependencies(
-        Dependency(
-          mod"org.scala-lang:scala-library",
+  protected def newIndex =
+    new DelegatingGlobalSymbolIndex(OnDemandSymbolIndex())
+
+  private def indexScalaLibrary(
+      index: GlobalSymbolIndex,
+      scalaVersion: String
+  ): Unit = {
+    val sources = Fetch
+      .create()
+      .withClassifiers(Set("sources").asJava)
+      .withDependencies(
+        Dependency.of(
+          "org.scala-lang",
+          "scala-library",
           // NOTE(gabro): we should ideally just use BuildoInfoVersions.scalaVersion
           // but using the 2.11 stdlib would cause a lot tests to break for little benefit.
           // We can remove this switch once we drop support for 2.11
-          BuildInfoVersions.scalaVersion match {
+          scalaVersion match {
             case v if v.startsWith("2.13") => v
             case v if v.startsWith("2.12") => v
             case _ => BuildInfoVersions.scala212
           }
         )
       )
-      .run()
+      .fetch()
+      .asScala
     sources.foreach { jar => index.addSourceJar(AbsolutePath(jar)) }
   }
 
-  def params(code: String, filename: String = "test.scala"): (String, Int) = {
+  private def newPresentationCompiler(scalaVersion: String): TestEnvironment = {
+
+    val binaryVersion =
+      ScalaVersions.scalaBinaryVersionFromFullVersion(scalaVersion)
+    val scalaDependencies =
+      if (ScalaVersions.isScala3Version(scalaVersion))
+        ScalaDependencies.scala3(scalaVersion)
+      else
+        ScalaDependencies.scala2(scalaVersion)
+
+    val scalaLibrary: Seq[Path] = Fetch
+      .create()
+      .withDependencies(scalaDependencies: _*)
+      .fetch()
+      .asScala
+      .map(_.toPath())
+
+    val extraClasspath: Seq[Path] =
+      Fetch
+        .create()
+        .withDependencies(extraDependencies(scalaVersion): _*)
+        .fetch()
+        .asScala
+        .map(_.toPath())
+
+    val extraScala211Classpath = ClasspathLoader
+      .getURLs(this.getClass.getClassLoader)
+      .filter(!_.toString().contains("scala"))
+      .map(url => Paths.get(url.toURI))
+
+    // pprint.log(extraScala211Classpath)
+
+    val myclasspath: Seq[Path] = (extraClasspath ++ scalaLibrary).filterNot {
+      path =>
+        val filename = path.toString()
+        filename.contains("scala-reflect") ||
+        filename.contains("scala-compiler") ||
+        filename.contains("scala-xml")
+    }
+
+    val index = newIndex
+    if (requiresJdkSources)
+      JdkSources().foreach(jdk => index.addSourceJar(jdk))
+    if (requiresScalaLibrarySources)
+      indexScalaLibrary(index, scalaVersion)
+    val indexer = new Docstrings(index)
+    val workspace = new TestingWorkspaceSearch
+    val search = new TestingSymbolSearch(
+      ClasspathSearch.fromClasspath(myclasspath),
+      new Docstrings(index),
+      workspace,
+      index
+    )
+
+    val scalacOpts = scalacOptions(myclasspath)
+
+    val pc = TestEnvironment
```

Comment:
Changed the way we run cross suite in order to be able to easily extend the existing tests without the need for having all test dependencies work in Scala 3. I now start a compiler for each version and create tests for all supported version.

As an added bonus, this way found a number of issues, which were missed because cross tests were not tested the same as mtags would eventually be used. This way is closer to that.